restli-server/src/test/java/com/linkedin/restli/internal/server/model/TestRestLiApiBuilder.java

```diff
@@ -198,4 +198,62 @@ public void testActionReturnType(Class<?> resourceClass, Class<?> expectedAction
       Assert.assertEquals(resourceMethodDescriptor.getActionReturnType(), expectedActionReturnType);
     }
   }
+
+  @DataProvider(name = "unsupportedFinderResourceTypeData")
+  private Object[][] unsupportedFinderResourceTypeData()
+  {
+    return new Object[][]
+        {
+            { FinderUnsupportedKeyUnstructuredDataResource.class, "KeyUnstructuredDataResource class does not support for @Finder methods" },
+            { FinderUnsupportedSingleUnstructuredDataResource.class, "SingleUnstructuredDataResource class does not support for @Finder methods" }
+        };
+  }
+
+  /**
+   * Ensures that when finder methods are processed, if the return type is not of a Record, then it will be warned.
+   * For instance, it should recognize that the "logical" return type for a method
+   * {@code Task<ActionResult<String>> doFoo();} is {@code String.class}.
+   *
+   * @param resourceClass resource used as an input
+   */
+  @Test(dataProvider = "unsupportedFinderResourceTypeData")
+  public void testFinderUnsupportedResourceType(Class<?> resourceClass, String expectedPartialMessage)
+  {
+    try
+    {
+      RestLiApiBuilder.buildResourceModels(Collections.singleton(resourceClass));
+      Assert.fail("For the finder resource class with a non RecordTemplate sub class, we shall throw an exception");
+    }
+    catch (ResourceConfigException resourceConfigException)
+    {
+      Assert.assertTrue(resourceConfigException.getMessage().contains(expectedPartialMessage),
+          String.format("Expected %s with message containing \"%s\" but instead found message \"%s\"",
+              ResourceConfigException.class.getSimpleName(), expectedPartialMessage, resourceConfigException.getMessage()));
+    }
+  }
+
+  @DataProvider(name = "finderSupportedResourceTypeData")
+  private Object[][] finderSupportedResourceTypeData()
+  {
+    return new Object[][]
+        {
+            { FinderSupportedAssociationDataResource.class },
+            { FinderSupportedComplexKeyDataResource.class },
+            { FinderWithActionResource.class }
+        };
+  }
+
+  @Test(dataProvider = "finderSupportedResourceTypeData")
+  public void testFinderSupportedResourceType(Class<?> resourceClass)
```

Comment:
You can use the expectedExceptions and expectedExceptionsMessageRegExp attributes of the @Test annotation

Reply:
This test is to ensure other resources classes other then the: [KeyUnstructuredDataResource, SingleUnstructuredDataResource] are not getting an exception, so the code in the test is not expecting exceptions. But the annotation of the two you suggested I will put them in my first test which there it expects exceptions.